### PR TITLE
feat: improve readme

### PR DESCRIPTION
**Pull request by Asmit2952**
Fixes: #83 

Added steps to install dependencies and locally set up the project. The readme now includes steps starting from installing Ruby to running project locally on localhost

The installation steps are covered in all major platforms: Ubuntu/Debian, Windows, macOS.

PS: Did these steps locally, please suggest changes if I am missing on something.

### CODE REVIEW

**Comment by Asmit2952** [comment]
I agree with the [Contributor License Agreement](https://www.semi.technology/playbooks/misc/contributor-license-agreement.html).

File: readme.md [hunk 1]
```diff
@@ -1,13 +1,95 @@
...
+  # set the ruby version locally to 2.7.5
+  rbenv local 2.7.5
+  ```
+
+- Check if dependencies are installed correctly
+  ```bash
+  ruby -v
+  gem -v
+  ```
+
```
Comment (bhavukkalra) [inline readme.md:60]: You might wanna add here.
`sudo apt-get install ruby-dev`
There is no ruby development environment installed. Development env is needed to compile ruby extensions, which are mostly written in C

File: readme.md [hunk 2]
```diff
@@ -1,13 +1,95 @@
...
+- Check if dependencies are installed correctly
+  ```bash
+  jekyll -v
+  ```
+### Setting up the repository
+  
+- To get the site up and running locally, follow the below steps:
+
+  **PS.:** You need to have a full Bash environment. If you're on Windows, please use WSL.
+
```
Comment (bhavukkalra) [inline readme.md:75]: Could you add the link to Microsoft Docs which tells how to activate WSL on windows here.
Link - [WSL](https://docs.microsoft.com/en-us/windows/wsl/install)

File: readme.md [hunk 3]
```diff
@@ -1,13 +1,95 @@
...
+- Create a local clone of the website:
+  ```
```
Comment (bhavukkalra) [inline readme.md:77]: I think its more preferable for incoming contributors to guide them to contribute via forks. Could you customize to incorporate a username based link like - `git clone git@github.com:[YOUR-USERNAME]/weaviate-io.git`.

Also add a pre step guiding the contributor to fork the repo first

File: readme.md [hunk 4]
```diff
@@ -1,13 +1,95 @@
...
+  rbenv local 2.7.5
+  ```
+  - For installing RubyGems, read [docs](https://jekyllrb.com/docs/installation/ubuntu/)
+  ```bash
+  echo '# Install Ruby Gems to ~/gems' >> ~/.bashrc
+  echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc
+  echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
+  source ~/.bashrc
+  ```
+
```
Comment (bhavukkalra) [inline readme.md:39]: Could you also add 
`export PATH="$HOME/.rbenv/shims:$PATH"`
due to this not being added I think the ruby version isn't getting picked up while doing `bundle install`.

**Review by bhavukkalra — CHANGES_REQUESTED** [review]

File: readme.md (hunk 4 above)
Comment (Asmit2952) [reply readme.md:39]: Nice catch!! Forgot to add this

File: readme.md (hunk 1 above)
Comment (Asmit2952) [reply readme.md:60]: https://www.mikekasberg.com/blog/2020/09/27/set-up-a-ruby-dev-environment.html

**Comment by laura-ham** [comment]
Looks great @Asmit2952, thanks!

**Review by laura-ham — COMMENTED** [review]
Looks great

**Pull request merged**